go/vt/vttablet/tabletmanager/rpc_replication.go

```diff
@@ -194,11 +192,6 @@ func (agent *ActionAgent) ResetReplication(ctx context.Context) error {
 
 // InitMaster enables writes and returns the replication position.
 func (agent *ActionAgent) InitMaster(ctx context.Context) (string, error) {
-	if err := agent.lock(ctx); err != nil {
```

Comment:
This looks unsafe. There's a lot happening now without holding the lock. The current goal of the lock is to serialize entire RPCs, not just certain changes. I don't think we can relax that without thinking through a full redesign of the locking scheme.

Reply:
+1